common/prometheus-server/templates/_helpers.tpl

```diff
@@ -135,3 +135,9 @@ prometheus-{{- (include "prometheus.name" .) -}}
   regex: ^({{ . | join "|" }})$
   action: keep
 {{- end -}}
+
+{{- define "prometheus.drop-metrics.metric-relabel-config" -}}
+- sourceLabels: [ __name__ ]
+  regex: ^({{ . | join "|" }})$
```

Comment:
List items should be escaped with `regexQuoteMeta` to avoid accidental overmatching or mismatching. Since the templating syntax does not have a `map` or similar function to apply the quoting across all list elements, I'd rather suggest passing in the forbidden metrics as one regex from the get-go.

Reply:
As per existing ServiceMonitor prometheus-collector-kubernetes-cadvisor, which uses the same syntax for rendering metricRelabelings, thei shoudn't be an issue:
```
    - action: drop
      regex: POD;(container_cpu_cfs_periods_total|container_cpu_cfs_throttled_seconds_total|container_cpu_cfs_throttled_periods_total|container_cpu_usage_seconds_total|container_cpu_user_seconds_total|container_cpu_system_seconds_total|container_fs_inodes_total|container_fs_limit_bytes|container_fs_usage_bytes|container_last_seen|container_memory_usage_bytes|container_memory_working_set_bytes|container_start_time_seconds)
      sourceLabels:
      - container
      - __name__
```